PerimeterXModule/Internals/ReverseProxy.cs

```diff
@@ -0,0 +1,240 @@
+﻿using PerimeterX;
+using System;
+using System.Collections.Specialized;
+using System.Diagnostics;
+using System.Net;
+using System.Text;
+using System.Web;
+
+namespace PerimeterX
+{
+	public interface IReverseProxy
+	{
+		void ReversePxClient(HttpContext context);
+		void ReversePxXhr(HttpContext context);
+		bool ShouldReverseClient(string uri);
+		bool ShouldReverseXhr(string uri);
+	}
+
+	public class ReverseProxy : IReverseProxy
+	{
+		private readonly string DEFAULT_CLIENT_VALUE = "";
+		private readonly string CONTENT_TYPE_JAVASCRIPT = "application/javascript";
+		private readonly string DEFAULT_JSON_VALUE = "{}";
+		private readonly string CONTENT_TYPE_JSON = "application/json";
+		private readonly byte[] DEFAULT_EMPTY_GIF_VALUE = { 0x47 , 0x49, 0x46, 0x38, 0x39, 0x61, 0x01, 0x00, 0x01, 0x00, 0x80, 0x00,
+		0x00, 0xff, 0xff, 0xff, 0x00, 0x00, 0x00, 0x2c, 0x00, 0x00, 0x00, 0x00,
+		0x01, 0x00, 0x01, 0x00, 0x00, 0x02, 0x02, 0x44, 0x01, 0x00, 0x3b };
+		private readonly string CONTENT_TYPE_GIF = "image/gif";
+
+		private readonly string XHR_PATH = "xhr";
+		private readonly string CLIENT_PATH = "init.js";
+
+		private string ClientReversePrefix;
+		private string XhrReversePrefix;
+		private string CollectorUrl;
+
+		public bool IsReusable
+		{
+			get { return true; }
+		}
+
+		private PxModuleConfigurationSection PxConfig { get; set; }
+
+		public ReverseProxy(PxModuleConfigurationSection pxConfig)
+		{
+			PxConfig = pxConfig;
+			string prefixFormat = "/{0}/{1}";
+			string appIdPrefix = pxConfig.AppId.Substring(2);
```

Comment:
This can be a setting that is stored on our config and is set only once when the module is started. Also no reason to do this (non-dynamic) substring each time.